### ISSUE TITLE
Rule Request: explicit rawValue for camelCased Codable String enum

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#2888]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Please describe the rule idea, format
this issue's title as `Rule Request: [Rule Name]` and describe:

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

We use `JSONDecoder.KeyDecodingStrategy.convertFromSnakeCase` in our project. Our enums are mostly snake case on the backend too. Because of that, some people in our team had mistakenly assumed that these enum cases would also be converted from snake case (which is not true).

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
enum Status: String, Codable {
   case ok // shouldn't trigger
   case notAcceptable // should trigger
   case maybeAcceptable = "maybe_acceptable" // shouldn't trigger
}

enum Status: String, Decodable {
   case ok // shouldn't trigger
   case notAcceptable // should trigger
   case maybeAcceptable = "maybe_acceptable" // shouldn't trigger
}

enum Status: String {
   case ok // shouldn't trigger
   case notAcceptable // shouldn't trigger
   case maybeAcceptable = "maybe_acceptable" // shouldn't trigger
}

enum Status: Int, Codable {
   case ok // shouldn't trigger
   case notAcceptable // shouldn't trigger
   case maybeAcceptable = -1 // shouldn't trigger
}
```

3. Should the rule be configurable, if so what parameters should be configurable?

Just the severity.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Opt-in, as this would only add noise in some projects

**Comment by marcelofabri** [comment]
Closed in #2889 💯
